src/Solcore/Frontend/TypeInference/TcStmt.hs

```diff
@@ -725,8 +798,8 @@ checkMeasure ps c
 
 -- type generalization
 
-generalize :: ([Pred], [Pred], Ty) -> TcM Scheme
-generalize (ps,qs,t)
+generalize :: ([Pred], Ty) -> TcM Scheme
```

Comment:
The current PR only changes the signature, but I have  a question about the body:

``` haskell
      (ps1,t1) <- withCurrentSubst (ps,t)
      t2 <- withCurrentSubst t1
      s <- getSubst
      let
          t3 = apply s t2
```

it seems that the current subst is applied three times to t. Why?